crates/vm/src/arch/extensions.rs

```diff
@@ -523,7 +529,7 @@ pub enum SystemPeriphery<F: PrimeField32> {
     Poseidon2(Poseidon2PeripheryChip<F>),
 }
 
-impl<F: PrimeField32> SystemComplex<F> {
+impl<F: PrimeField32 + TwoAdicField> SystemComplex<F> {
```

Comment:
let's remove this and try to use a setter. I think it complicates the structs / traits too much to add `TwoAdicField` everywhere